excel/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java

```diff
@@ -60,14 +61,47 @@ public static void register(ReaderRegistry registry) {
 
   @Override
   public Table read(XlsxReadOptions options) throws IOException {
-    List<Table> tables = readMultiple(options);
+    List<Table> tables = readMultiple(options, true);
     if (tables.isEmpty()) {
       throw new IllegalArgumentException("No tables found.");
     }
-    return tables.get(0);
+    if (options.sheetIndex() != null) {
+      if (options.sheetIndex() >= 0 && options.sheetIndex() < tables.size()) {
+        int index = options.sheetIndex();
+
+        Table table = tables.get(index);
+        if (table == null) {
+          throw new IllegalArgumentException("No tables found.");
+        }
+        return table;
+      } else {
+        throw new IndexOutOfBoundsException(
+            "Sheet index outside bounds. " + tables.size() + " sheets found.");
+      }
+    }
+    // since no specific sheetIndex asked, return first table
+    Optional<Table> table = tables.stream().filter(t -> t != null).findFirst();
+    if (!table.isPresent()) {
+      throw new IllegalArgumentException("No tables found.");
+    } else {
+      return table.get();
+    }
   }
 
   public List<Table> readMultiple(XlsxReadOptions options) throws IOException {
+    return readMultiple(options, false);
+  }
+
+  /**
+   * Read at most a table from every sheet.
+   *
+   * @param options
```

Comment:
should either fill out or remove. probably the same with the `@throws` documentation